core/tests/unit/test_dms.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+
+from aws_cdk.assertions import Template
+from aws_ddk_core.base import BaseStack
+from aws_ddk_core.resources import DMSFactory, S3Factory
+
+
+def test_s3_source_endpoint(test_stack: BaseStack) -> None:
```

Comment:
The two tests below are pretty identical, so I would reduce it to one and parameterise with:
```
@pytest.mark.parametrize("type", ["source", "target"])
```

Reply:
Awesome suggestion, thanks. TIL